include/ginkgo/core/base/dim.hpp

```diff
@@ -200,11 +215,154 @@ struct dim<1u, DimensionType> {
         return dim(x.first_ * y.first_);
     }
 
+    friend GKO_ATTRIBUTES std::ostream &operator<<(std::ostream &os,
+                                                   const dim &x)
+    {
+        os << x.first_;
+        return os;
+    }
+
 private:
     dimension_type first_;
 };
 
 
+/**
+ * A type representing the dimensions of a multidimensional batch object.
+ *
+ * @tparam Dimensionality  number of dimensions of the object
+ * @tparam DimensionType  datatype used to represent each dimension
+ *
+ * @ingroup batch_dim
+ */
+template <size_type Dimensionality = 2, typename DimensionType = size_type>
+struct batch_dim {
+    static constexpr size_type dimensionality = Dimensionality;
+    using dimension_type = DimensionType;
+
+    /**
+     * Checks if the batch_dim object stores equal sizes.
+     *
+     * @return bool representing whether equal sizes are being stored
+     */
+    bool stores_equal_sizes() const { return equal_sizes_; }
+
+    /**
+     * Get the number of batches stored
+     *
+     * @return num_batches
+     */
+    size_type get_num_batches() const { return num_batches_; }
+
+    /**
+     * Get the batch sizes as a std::vector.
+     *
+     * @return  the std::vector of batch sizes
+     */
+    std::vector<dim<dimensionality, dimension_type>> get_batch_sizes() const
```

Comment:
```suggestion
    std::vector<dim<dimensionality, dimension_type>> get_sizes() const
```
We could also use `get_entry_sizes` if you want to be more specific.

Reply:
I think here batch_sizes is appropriate as you are getting the sizes of all the batches in a std::vector.

Reply:
It should be fine. I'm just saying that there's only one batch here, not many batches. There are many entries in the batch. But alright, I think in this case get_batch_sizes works regardless.